lib/smartthings.js

```diff
@@ -66,7 +66,13 @@ module.exports = class SmartThings {
     }
 
     setInputSource(value) {
-        return this.sendCommands({component: 'main', capability: 'mediaInputSource', command: 'setInputSource', arguments: [value]});
+        // Try to set input source with Samsung-specific vendor extended capability (used in e.g. smart monitors)
+        return this.sendCommands({component: 'main', capability: 'samsungvd.mediaInputSource', command: 'setInputSource', arguments: [value]})
```

Comment:
I cannot merge this PR because of this.

The plugin is developed for TVs, not for Smart Monitors. So the main focus should be for TVs.

Also, this function should not catch errors.

If this is the implementation that helps you, please fork the plugin and change the code as it fits your specific needs :)

Reply:
Could you please show me a dump of `http://TV_IP:8001/api/v2/` from your monitor? Maybe there is something we can relate to and change the capability based on that.